spectral_cube/spectral_cube.py

```diff
@@ -2328,6 +2328,186 @@ def _new_cube_with(self, **kwargs):
 
     _new_cube_with.__doc__ = BaseSpectralCube._new_cube_with.__doc__
 
+    def spatial_smooth_median(self, ksize,
+                        #numcores=None,
+                        **kwargs):
+        """
+        Smooth the cube along the spectral dimension
+
+        Parameters
+        ----------
+        kernel : `~astropy.convolution.Kernel1D`
+            A 1D kernel from astropy
+        convolve : function
+            The astropy convolution function to use, either
+            `astropy.convolution.convolve` or
+            `astropy.convolution.convolve_fft`
+        kwargs : dict
+            Passed to the convolve function
+        """
+
+        shape = self.shape
+
+        # "cubelist" is a generator
+        # the boolean check will skip smoothing for bad spectra
+        # TODO: should spatial good/bad be cached?
+        cubelist = ((self.filled_data[ii],
```

Comment:
this is a list of 2D images, no?  so `imagelist` or `planelist` would be better?

Reply:
fixed and push coming shortly